integration/test_devices.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+
+
+@pytest.mark.skip("Failing with 403.")
+def test_block(connection):
+    response = connection.devices.block(891804948033320117)
+    assert response.status_code == 201
+
+
+@pytest.mark.skip("Failing with 404.")
+def test_get_agent_full_disk_access_state(connection):
+    response = connection.devices.get_agent_full_disk_access_state(891804948033320117)
+    assert response.status_code == 200
+
+
+def test_get_by_guid(connection):
+    response = connection.devices.get_by_guid(891804948033320117)
+    assert response.status_code == 200
+
+
+def test_get_settings(connection):
```

Comment:
This test passes intermittently, at times returning `Py42Response`, 
`>>> sdk.devices.get_settings(891804948033320117)
<Py42Response [status=200, data={'SUPPORT_ACCESS': {'id': '2686', 'locked': False, 'scope': 'ORG', 'value': 'false'}, 'c42.msa.acceptance': {'id': '2686', 'locked': True, 'scope': 'ORG', 'value': '890854304274...T22:22:53:922'}, 'c42_indexing_enabled': {'locked': False, 'scope': 'SYSTEM', 'value': 'false'}, 'c42_indexing...ts_exclusions': {'locked': False, 'scope': 'SYSTEM', 'value': '3dm,3ds,3g2,...uv,z,zip,zipx'}, ...}]>`

otherwise `<DeviceSettings: guid: 891804948033320117, name: DESKTOP-3DG3V50>`. and hence assert statement fails.

Reply:
Weird.. The return type is expected to be a `DeviceSettings` object.

The method `update_settings()` however does return a `Py42Response`.

This guides explains it more: https://py42docs.code42.com/en/stable/userguides/devicesettings.html

Were you perhaps calling update when you got the `Py42Response`? I am not seeing how in the code it could happen intermittently; I find that really peculiar! Can you dig in more?

Reply:
@timabrmsn knows most about this feature as well

Reply:
Are we testing older versions of py42? `get_settings()` used to just call the [DeviceSetting](https://www.crashplan.com/apidocviewer/#DeviceSetting) endpoint, which is what that Py42Response result looks like.

Reply:
1.9.0 is when this behavior changed.

Reply:
changed.